### PR TITLE
Kyriv/refactor solution transfers

### PR DESCRIPTION
**Pull request by kyriv1980**
Pr that creates the new MeshlessInterpolationFunction class and refactors as needed.

### CODE REVIEW

**Comment by kyriv1980** [comment]
@roystgnr I re-did the Pr from the beginning cause i didn't want to get into the weeds with git history etc...

**Comment by roystgnr** [comment]
This looks good.  Only one change I can see to make, but let's wait until we get the coverage report for this version of the branch first.

File: src/solution_transfer/meshfree_solution_transfer.C
Comment (roystgnr) [inline src/solution_transfer/meshfree_solution_transfer.C:61]: We should probably add the above 5 move/copy/assignment methods to the new consolidated class.  If @jwpeterson was enough of a stickler to put them in this internal-library-use-only version, I can't imagine him letting us get away with leaving them out of the exposed version.

**Review by roystgnr — CHANGES_REQUESTED** [review]

File: include/solution_transfer/meshless_interpolation_function.h
Comment (jwpeterson) [inline include/solution_transfer/meshless_interpolation_function.h:39]: Don't delete this space.

File: include/solution_transfer/meshless_interpolation_function.h
Comment (jwpeterson) [inline include/solution_transfer/meshless_interpolation_function.h:42]: Or this one.

**Review by jwpeterson — CHANGES_REQUESTED** [review]
Please keep all the spacing the way it was originally. There's no reason to have any whitespace only diffs in this file, that way we can focus on things which actually changed.

File: include/solution_transfer/meshless_interpolation_function.h
Comment (kyriv1980) [reply include/solution_transfer/meshless_interpolation_function.h:39]: at some point i deactivated the clang formatting that edits the code when i save. But maybe some edits got through before that. Is there something equivalent for the libmesh world?

File: include/solution_transfer/meshfree_solution_transfer.h
Comment (jwpeterson) [inline include/solution_transfer/meshfree_solution_transfer.h:21]: Although It's not part of your change, I just noticed that this include guard doesn't follow our usual pattern. So, these two lines could be changed to:
```
#ifndef MESHFREE_SOLUTION_TRANSFER_H
#define MESHFREE_SOLUTION_TRANSFER_H
```
if you felt so inclined.

File: include/solution_transfer/meshless_interpolation_function.h
Comment (jwpeterson) [reply include/solution_transfer/meshless_interpolation_function.h:39]: I think @lindsayad was going to throw a .clang-format file into the libmesh repo at some point, but it doesn't look like he did so yet. Other than that, we just have some [formatting conventions](https://github.com/libMesh/libmesh/wiki/Source-code-formatting-conventions) that we use, but aren't rigidly enforced by anything.

Comment (lindsayad) [reply include/solution_transfer/meshless_interpolation_function.h:39]: Yea I haven't thrown anything in yet, to my chagrin

Comment (roystgnr) [reply include/solution_transfer/meshless_interpolation_function.h:39]: I tried to throw in one once, but it kind of backfired on me as soon as I started saving files with my vim "auto-apply .clang-format" settings.  clang-format doesn't have (and IIRC has a FAQ about not having) an option to tell it to enforce rules A1, B3, and C2 while simply preserving the existing formatting for rules D through Z; it's an all-or-nothing reformatter.

If we want to decide on formatting for every option they have, I guess we could do all the changes in one giant auto-reformat-everything commit and then proceed from there, but that's the only way to avoid editors turning every `:wq` into a giant mess of whitespace changes.

**Comment by roystgnr** [comment]
John asked me about something that I agree we might want to nip in the bud - since this is the first time we're exposing this class to users, let's fix up the name consistency while we can still do so without breaking anybody's code?  If we replace `s/MeshlessInterpolationFunction/MeshfreeInterpolationFunction/g`, and similarly on all the `meshless_` filenames and `MESHLESS_` macros, that'll match the terminology choice in the existing `MeshfreeInterpolation` and `MeshfreeSolutionTransfer` classes.

**Review by jwpeterson — APPROVED** [review]
This looks much better. I have one or two very minor formatting nitpicks still but I can just fix those before merge.

File: include/solution_transfer/meshless_interpolation_function.h
Comment (lindsayad) [reply include/solution_transfer/meshless_interpolation_function.h]: I tell emacs to exclude certain directories like emacs for my clang formatting hook. I would assume you could tell vim the same thing, but if not, then I guess emacs must be better 😺

**Review by roystgnr — APPROVED** [review]

**Comment by roystgnr** [comment]
Looks good to me; I'll leave it to John to hit the button after he pushes whatever formatting tweaks to the branch.  Thanks!

File: include/solution_transfer/meshless_interpolation_function.h
Comment (roystgnr) [reply include/solution_transfer/meshless_interpolation_function.h]: :fishing_pole_and_fish:

I can tell vim to ignore libMesh directories, but then what's the point of having a .clang-format file there?  Just to look pretty?

Comment (lindsayad) [reply include/solution_transfer/meshless_interpolation_function.h]: What I do when developing new code in libMesh is to write it, then select the new code region, and then `M-x clang-format`

**Pull request merged**
